blogposts/2021/how-to-deconstruct-the-monolith.md

```diff
@@ -0,0 +1,102 @@
+
+### The branch by abstraction pattern
+
+When we can’t easily intercept calls and create a proxy to reroute the requests, we can benefit from the branch by abstraction technique. It’s like a strategy design pattern applied to a more extensive scope.
+
+Branch by abstraction requires us to modify the existing monolith’s code. It works by implementing the next few steps:
+
+1. Identify the code within the monolith that composes the functionality. This means finding out classes, methods, and parameters without worrying about the actual implementation. Finding those functional lines in a vast monolith codebase is often an arduous task. To make the lives of your microservices developers easier, consider [universal code search](https://about.sourcegraph.com/). It provides a search engine for code across all your repositories and branches, so you can find and filter algorithms, definitions, and references instantaneously.
+2. Create the interfaces that compose the functionality’s abstraction. This means creating a new code that defines a new “contract” for classes, methods, and parameters discovered in the previous step. 
+3. Refactor the old monolith’s code, ensuring it adheres to the interfaces defined by the new abstraction. Notice that this step doesn’t break or change the existing functionality in any way but decouples the identified functionality from the rest of the monolith. 
+4. Implement the new service, ensuring the existing functionality behaves precisely as in the monolith.
+5. Create feature flags that allow you to easily switch between the old monolith functionality and the new implementation based on the same abstraction.
+
+Finally, once the new microservice abstraction of the functionality is stable, we must deprecate the old monolith’s functionality. Later, to avoid confusion, we’ll remove it entirely from our source code.
+
+## Conclusion
+
+Migrating from a monolithic architecture to microservices gives us many benefits. These include scalability and the ability to assign many developer teams to build different services simultaneously.
+
+Modernizing a monolithic application is a multi-team development effort. It takes planning and deliberate work to break a monolith into smaller parts. Our tour described how to leverage tools, patterns, and methods tested in the field to guide our efforts towards a safe and consistent microservices development. So, when it’s time to say goodbye to your monolith, you have the knowledge you need to prepare for a successful migration.
```

Comment:
```suggestion
Modernizing a monolithic application is a multi-team development effort. It takes planning and deliberate work to break a monolith into smaller parts. Our tour described how to leverage tools, patterns, and methods tested in the field to guide our efforts towards a safe and consistent microservices development. So, when it’s time to say goodbye to your monolith, you have the knowledge you need to prepare for a successful migration.
```
**About the author** \
Marcelo Ricardo de Oliveira is a senior freelance software developer who lives with his lovely wife Luciana and his little buddy and stepson Kauê in Guarulhos, Brazil. He is the co-founder of the Brazilian TV Guide TV Map and currently works for Alura Cursos Online.

Reply:
For some reason I couldn't apply this as a suggestion, done here instead https://github.com/sourcegraph/about/pull/3369/commits/f8e231e5d5dc4d89a861798faacd6426a9052e51